pkg/cluster/install.go

```diff
@@ -283,7 +283,7 @@ func (m *manager) bootstrap() []steps.Step {
 		steps.Action(m.ensureServiceEndpoints),
 		steps.Action(m.setMasterSubnetPolicies),
 		steps.AuthorizationRetryingAction(m.fpAuthorizer, m.deployBaseResourceTemplate),
-		steps.Condition(m.attachNSGs, 3*time.Minute, true),
+		steps.AuthorizationRetryingAction(m.fpAuthorizer, m.attachNSGs),
```

Comment:
To check my understanding: This changes from
- retry on `m.attachNSGs` failure for 3 mins
to
- run `m.attachNSGs` (and in it we run for 3 mins with 30s intervals)

If so why is this better? Fewer calls is not nothing but it's not clear to me how this addresses the main points you mentioned.

Reply:
It's possible for the NSGs to not pass the authorisation checks immediately, so we need to also refresh that -- the existing code doesn't do that.